From 3e7939e08a47250fdaa16d5f49f5343a79392cd4 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 12 Feb 2012 19:01:43 -0800
Subject: [PATCH] nls: Use LC_CTYPE=C for sed magic

Stricter versions of sed can trip up if the input does not match
LC_CTYPE

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 cpprules.in     |    2 +-
 nls/Makefile.am |    6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cpprules.in b/cpprules.in
index fca0ab3..e1296c6 100644
--- a/cpprules.in
+++ b/cpprules.in
@@ -2,7 +2,7 @@
 # Rules for generating files using the C pre-processor
 # (Replaces CppFileTarget from Imake)
 
-SED = sed
+SED = LC_CTYPE=C sed
 
 SUFFIXES += .pre
 
diff --git a/nls/Makefile.am b/nls/Makefile.am
index 7dc4308..0eced4c 100644
--- a/nls/Makefile.am
+++ b/nls/Makefile.am
@@ -20,19 +20,19 @@ include $(top_srcdir)/cpprules.in
 
 locale.alias: locale.alias.pre
 	$(AM_V_GEN)$(RAWCPP) $(RAWCPPFLAGS) $(CPP_FILES_FLAGS) < $(srcdir)/locale.alias.pre | $(CPP_SED_MAGIC) > locale.alias.l1
-	sed -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
+	$(SED) -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
         < locale.alias.l1 > locale.alias.l2
 	cat locale.alias.l2 locale.alias.l1 > locale.alias
 
 compose.dir: compose.dir.pre
 	$(AM_V_GEN)$(RAWCPP) $(RAWCPPFLAGS) $(CPP_FILES_FLAGS) < $(srcdir)/compose.dir.pre | $(CPP_SED_MAGIC) > compose.dir.l1
-	sed -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
+	$(SED) -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
         < compose.dir.l1 > compose.dir.l2
 	cat compose.dir.l2 compose.dir.l1 > compose.dir
 
 locale.dir: locale.dir.pre
 	$(AM_V_GEN)$(RAWCPP) $(RAWCPPFLAGS) $(CPP_FILES_FLAGS) < $(srcdir)/locale.dir.pre | $(CPP_SED_MAGIC) > locale.dir.l1
-	sed -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
+	$(SED) -e '/^[^#][^        ]*:/s/://' -e '/^[^#].*[        ].*:/d' \
         < locale.dir.l1 > locale.dir.l2
 	cat locale.dir.l2 locale.dir.l1 > locale.dir
 
-- 
1.7.7.5 (Apple Git-28)

